examples/OpenTracing.Examples/ListenerPerRequest/Client.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OpenTracing.Tag;
+
+namespace OpenTracing.Examples.ListenerPerRequest
+{
+    public class Client
+    {
+        private readonly ITracer _tracer;
+
+        public Client(ITracer tracer)
+        {
+            _tracer = tracer;
+        }
+
+        // Async execution
+        private Task<String> Execute(String message, ResponseListener responseListener)
+        {
+            return Task.Run(() =>
```

Comment:
Also looks very non-C#y :smile: How about just adding a "await Task.Delay(10)" to force the async execution?